packages/ui/src/ui/store/actions/system/masters.ts

```diff
@@ -129,10 +122,35 @@ async function loadMastersConfig(): Promise<[MastersConfigResponse, MasterAlert[
         },
     ];
 
-    const [primaryMasterCellTag, timestampProviderCellTag] = await ytApiV3Id.executeBatch(
-        YTApiId.systemMastersConfig,
-        {requests: [...tag_cell_requests]},
-    );
+    const [timestampProviderCellTag] = await ytApiV3Id.executeBatch(YTApiId.systemMastersConfig, {
+        requests: [...timestamp_tag_cell_requests],
+    });
+
+    let primaryMasterCellTag: BatchResultsItem<any> | undefined;
+
+    for (const primary_path of primaryMasterPaths) {
+        const tag_cell_requests = [
+            {
+                command: 'get' as const,
+                parameters: {
+                    path: `//sys/primary_masters/${primary_path}/orchid/config/primary_master/cell_id`,
+                    ...USE_SUPRESS_SYNC,
+                },
+            },
+        ];
+
+        [primaryMasterCellTag] = await ytApiV3Id.executeBatch(YTApiId.systemMastersConfig, {
```

Comment:
Instead of doing requests sequentially it is better to prepare batch-request to all primary-master and take first available output.

Reply:
Good idea, done!